cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/LdapResourceCRLFetcher.java

```diff
@@ -0,0 +1,181 @@
+import java.security.cert.CertificateException;
+import java.security.cert.X509CRL;
+import java.util.Hashtable;
+import java.util.Map;
+
+/**
+ * Fetches a CRL from an LDAP instance.
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public class LdapResourceCRLFetcher extends ResourceCRLFetcher {
+
+    /** The Certificate revocation list attribute name.*/
+    protected final String certificateRevocationListAttributeName;
+
+    /** Authentication mode for ldap. **/
+    protected String securityAuthentication;
+
+    /** Authentication principal for ldap. **/
+    protected String securityPrincipal;
+
+    /** Authentication credentials for ldap. **/
+    protected String securityCredentials;
+
+    /** object name to look up in the context. **/
+    protected String objectName;
+
+    /**
+     * Instantiates a new Ldap resource cRL fetcher.
+     * initializes the default certificate revocation attribute
+     * to be {@link CertUtils#DEFAULT_CERTIFICATE_REVOCATION_LIST_ATTRIBUTE}.
+     */
+    public LdapResourceCRLFetcher() {
+        this(CertUtils.DEFAULT_CERTIFICATE_REVOCATION_LIST_ATTRIBUTE);
+    }
+
+    /**
+     * Instantiates a new Ldap resource cRL fetcher.
+     *
+     * @param certificateRevocationListAttributeName the certificate revocation list attribute name
+     */
+    public LdapResourceCRLFetcher(final String certificateRevocationListAttributeName) {
+        this.certificateRevocationListAttributeName = certificateRevocationListAttributeName;
+    }
+
+    /**
+     * This can be one of the following strings:
+     * "none", "simple", sasl_mech, where sasl_mech is a
+     * space-separated list of SASL mechanism names.
+     * @param securityAuthentication the authn mode
+     */
+    public void setSecurityAuthentication(final String securityAuthentication) {
+        this.securityAuthentication = securityAuthentication;
+    }
+
+    public void setSecurityPrincipal(final String securityPrincipal) {
+        this.securityPrincipal = securityPrincipal;
+    }
+
+    public void setSecurityCredentials(final String securityCredential) {
+        this.securityCredentials = securityCredential;
+    }
+
+    public void setObjectName(final String objectName) {
+        this.objectName = objectName;
+    }
+
+    @Override
+    protected X509CRL fetchInternal(final Object r) throws Exception {
+        if (r.toString().toLowerCase().startsWith("ldap")) {
+            return fetchCRLFromLdap(r);
+        }
+        return super.fetchInternal(r);
+    }
+
+    /**
+     * Downloads a CRL from given LDAP url, e.g.
+     * <code>ldap://ldap.infonotary.com/dc=identity-ca,dc=infonotary,dc=com</code>
+     *
+     * @param r the resource
+     * @return the x 509 cRL
+     * @throws Exception if connection to ldap fails, or attribute to get the revocation list is unavailable
+     */
+    protected X509CRL fetchCRLFromLdap(final Object r) throws Exception {
+        DirContext ctx = null;
```

Comment:
@serac would you be amenable to me merging this pull? and then you could take over the ldaptive config when time allows? It's taking me longer than I'd hoped and I'd be happy to pass it over to you. 

Or if you'd rather make changes to this pull, that'd be fine too.